drivers/stm32_eth/stm32_eth.c

```diff
@@ -135,9 +152,28 @@ static int _get(netdev_t *dev, netopt_t opt, void *value, size_t max_len)
     return res;
 }
 
+#if IS_USED(MODULE_STM32_ETH_LINK_UP)
+static void _timer_cb(void *arg)
+{
+    netdev_t *dev = (netdev_t *)arg;
+    if (_get(dev, NETOPT_LINK, NULL, 0)) {
+        link_status = 1;
+        dev->event_callback(dev, NETDEV_EVENT_ISR);
```

Comment:
Why not call `netdev->event_callback(netdev, NETDEV_EVENT_LINK_UP);` here directly?

Reply:
Because the timer fires in ISR so we first have to dispatch out of it.

Reply:
Because callbacks from `xtimer_set` are being executed in an interrupt context. Otherwise I get a `Stack pointer corrupted` error message.